packages/fxa-settings/src/pages/Signin/SigninTotpCode/container.tsx

```diff
@@ -102,6 +127,22 @@ export const SigninTotpCodeContainer = ({
 
       // Check authentication code
       if (result.data?.verifyTotp.success === true) {
+        // Attempt to finish any pending key stretching upgrade. We cannot finalize
+        // a key stretching upgrade until the session is verified, which the process
+        // can only be finished after the account has been verified on accounts that
+        // require totp.
+        if (signinState?.sessionToken) {
+          await tryFinalizeUpgrade(
```

Comment:
nit: do we need to await?  it has no effect on the return value and there's error handling in the function.

Reply:
I think we need the await. The next thing that happens is typically navigation, and I think that could interrupt the upgrade process.